src/test/java/hudson/plugins/ec2/AmazonEC2CloudTest.java

```diff
@@ -92,7 +92,14 @@ public void testSshKeysCredentialsIdRemainsUnchangedAfterUpdatingOtherFields() t
         HtmlTextInput input = form.getInputByName("_.cloudName");
 
         input.setText("test-cloud-2");
-        r.submit(form);
```

Comment:
For this test, `r.submit(form)` returns 
```
FailingHttpStatusCode 404 Not Found for http://localhost:65397/jenkins/cloud/us-east-1/
```
I'm not sure if this is a regression, but it looks like because we change the name of the cloud node, the url is now dead. If this is acceptable, I did not want to add an `assertThrows` in case changes in the core would make that error go away.

Reply:
Sorry, @car-roll I did not fully understood your comment. From what you say it looks like changing the cloud name now is broken? If not what is being done by the production code to deal with that 404 that now is happening? It looks to me that instead of a try catch we should prevent that error to happen in first case

Reply:
Ouch, yeah so I manually tested this and this is a regression. Changing the names in the cloud config page fails on 404 😬

Reply:
yes, with a dedicated rename page